flink-table-store-spark/src/main/java/org/apache/flink/table/store/spark/SparkCatalog.java

```diff
@@ -201,7 +202,9 @@ public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceExcepti
     @Override
     public SparkTable loadTable(Identifier ident) throws NoSuchTableException {
         try {
-            return new SparkTable(catalog.getTable(objectPath(ident)));
+            ObjectPath path = objectPath(ident);
+            return new SparkTable(
+                    catalog.getTable(path), Lock.factory(catalog.lockFactory().orElse(null), path));
```

Comment:
When would the lock factory of the catalog be null? Could this directly use `CatalogLockFactory`?

Reply:
I want to expose only a `Lock` to `SparkTable`, instead of database and table name. So convert Catalog Lock to Lock.

Reply:
If the catalog can not provide a lock, for example, `FileSystemCatalog` can not provide a lock.